bundles/map-editor/src/main/java/org/orbisgis/mapeditor/map/MapControl.java

```diff
@@ -430,11 +430,13 @@ protected Object doInBackground() throws Exception {
             } catch (Exception ex) {
                 LOGGER.error(ex.getLocalizedMessage(), ex);
             } finally {
-                mapControl.getMapTransform().setImage(rendererImage);
-                intermediateDrawing.set(false);
-                awaitingDrawing.set(false);
-                updateViewTime.stop();
-                mapControl.repaint();
+                if(!isCancelled()) {
```

Comment:
Warning, if cancel is ongoing and awaitingDrawing is true then no further drawing could be done.